pkg/ring/ring.go

```diff
@@ -56,19 +56,17 @@ var ErrEmptyRing = errors.New("empty ring")
 
 // Config for a Ring
 type Config struct {
-	Consul            ConsulConfig  `yaml:"consul,omitempty"`
-	Store             string        `yaml:"store,omitempty"`
+	KVStore           KVConfig      `yaml:"kvstore,omitempty"`
 	HeartbeatTimeout  time.Duration `yaml:"heartbeat_timeout,omitempty"`
 	ReplicationFactor int           `yaml:"replication_factor,omitempty"`
 
 	Mock KVClient
 }
 
-// RegisterFlags adds the flags required to config this to the given FlagSet
+// RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
-	cfg.Consul.RegisterFlags(f)
+	cfg.KVStore.RegisterFlags(f, "")
```

Comment:
Should this be "ring"?

Reply:
Hmmm actually no it shouldn't.  The flags were `consul...` before, and after this change shouldn't be `ring.consul...` or you'll break all the previous deployments.